### ISSUE TITLE
Object is possibly 'undefined' when accessing arrays via at

### ISSUE THREAD

**Issue by dandv** [microsoft/TypeScript#57224]
### 🔎 Search Terms

"Object is possibly 'undefined' array at"

### 🕗 Version & Regression Information

Tested with v5.3.3 in the Playground, with the target set to ES2022.

### ⏯ Playground Link

https://www.typescriptlang.org/play?target=9#code/MYewdgzgLgBADiAlmKEYF4YG0BQMYDee+MAZiCAFwwAMANMQL4MC6A3DjqJCADYCmAOl4gA5gAoEyVFhotB5EAEoO3CHyEiJUlBCw7Uw-mFFQAFgFoAjPMUqu4dQOFjJSXYICGUcTSUKKFXwAemCYZFAAJ0j+YFgAd09IsGRRagB5ACMAK1jYRDQECAhETN4ATxgAcgBXMAATflJkfnqqhx5nLTdpCC8fa387NhCwiJBovJhE5NSMnKmC+BBi0orqusbmsFb2tQ0XbXcZKwAmAGYAFltAkZhQmEzK828YLURgOmmzSrAQWFE-Fg5n4MAgngAtqCZikTDAzPwYgB+HBAA

### 💻 Code

```ts
const points = [
  {
    foo: 'bar',
  },
];

console.log(points[0].foo);  // ok, no warning
console.log(points[points.length-1].foo);  // ok, no warning
console.log(points.at(0).foo);  // incorrect warning: Object is possibly 'undefined'
console.log(points.at(-1).foo);  // incorrect warning: Object is possibly 'undefined'
console.log(points[1234].foo);  // by that logic, why not get the same warning here?
```

### 🙁 Actual behavior

Incorrect warning: `Object is possibly 'undefined'` for `array.at(...).property);`

### 🙂 Expected behavior

No warning for accessing array elements via [`.at`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at), for the same reason there's no warning when accessing via index notation.

**Comment by fatcerberus** [comment]
All of those are errors under `noUncheckedIndexedAccess`.  `.at()` is just strict by default because there's no means to change the type of the function depending on compiler options.

**Comment by maurer2** [comment]
Hello, I noticed that the type narrowing doesn't seem to work with comparisons that involve greater and smaller signs even when checking the type beforehand. Is it possible to do this without resorting to a custom type guard? 

Listing:
```typescript
const list = [1, 2, 3, 4, 5];

const islastItemPositive = typeof list.at(-1) === 'number' && list.at(-1) > 0;
const islastItemNegative = typeof list.at(-1) === 'number' && list.at(-1) < 0;
const islastItemZero = typeof list.at(-1) === 'number' && list.at(-1) === 0;
const islastItemNotZero = typeof list.at(-1) === 'number' && list.at(-1) !== 0;
```

The TS compiler complains that `list.at(-1) > 0` and `list.at(-1) < 0;` are potentially undefined. it doesn't show that error message for the last two lines.

<img width="1251" alt="image" src="https://github.com/user-attachments/assets/5972ed8c-3c83-4e5a-9820-09f1fa04ce3c">


I recreated the example in TS playground: https://www.typescriptlang.org/play/?target=10&ts=5.5.4#code/MYewdgzgLgBANgS2jAvDA2gRgDQwEy4DMuALLgKwC6A3AFC2iSxJwCG0AklAKYC2ACiAgIoCAG7dUMKAE8ADtxAAzeEigA6VlAAUAWkwBKVCjQByMAFdeAI24AnUzABkT1dE079RgHwwADHSMyCzsUFx8AHLcAOZa4pJosgrKbhpaeobGZpY29o4uqR4ZRgA8-oHgwRBsnDy8AFr2IFJJiiqI7uleWTDmVrYOzq4daZ6ZJmgBDJXM1aHhvBEgUI12zYnybYVd4ya9OQP5w2pF3QCEewFAA

Cheers
